src/script/interpreter.cpp

```diff
@@ -1514,6 +1514,8 @@ void PrecomputedTransactionData::Init(const T& txTo, std::vector<CTxOut>&& spent
         m_spent_outputs_ready = true;
     }
 
+    // TODO: Improve this heuristic
+    bool uses_bip119_ctv = true;
```

Comment:
But my question is: why is the cache apparently not required for the segwit code I link to there? And when you say, "required," you just mean to prevent quadratic hashing, right?

Reply:
it *is* required. E.g., we could (and maybe _should_) add an assert for something like:

```c++
if (!evaulting_for_transaction_signing) assert(cacheReady)
```

But it's not "required" strictly speaking because the code is designed to be usable from both tx signing logic and verification. This is perhaps bad API design and these could be separated into a consensus and non consensus version. honestly... it might even be an issue that we expose the signing logic to DoS (but you just... DoS yourself trying to sign a PSBT?) if we don't cache this stuff.